server/routes/posts.js

```diff
@@ -0,0 +1,118 @@
+const express = require("express");
+const router = express.Router();
+const Post = require("../models/Post.js");
+const { response } = require("express");
+const { update } = require("../models/Post.js");
+
+//get all public posts for All/Explore page
+router.get('/', async (req, res, next) => {
+    try {
+        const posts = await Post.find({isPublic: true}).exec();
+        res.status(200).send(posts);
+    } catch(err) {
+        res.status(500).send(err);
+    }
+});
+
+//get post by id (pinnedPosts under User)
+router.get('/:id', async (req, res, next) => {
+    try {
+        const id = req.params.id;
+        const post = await Post.findById(id).exec();
+        res.send(post);
+    } catch(err) {
+        res.status(500).send(err);
+    }
+});
+
+//create a new post, will need to get type from user
+router.post('/', async (req, res, next) => {
+    try {
+        const post = Post(req.body);
+        const exists = await Post.findOne(post, "-_id -timestamp").exec();
+        console.log(exists);

```

Comment:
This prob can be done in a separate PR, but i think for best user experience, when the user wants to create a Youtube post or a github post, it'd be nice if they only need to input the youtube link or the github repo & issue#. So we'd need to make a call to their respective APIs. I think it'd be best to do it in the backend rather than make a separate call in the frontend before submitting the post.